Automattic/wp-calypso
# update: account password data

#### Changes proposed in this Pull Request

* Remove the `AccountPasswordData` event emitter
* Split up `AccountPasswordData` and create a general `generatePassword` utility function
* Move password validation to `me/account-password`
* Make `AccountPassword` a class and remove `createReactClass` 

#### Testing instructions

* Go to `/me/security/password` or `/me/security` and choose _Password_
* Make sure proper validation messages show up in the following cases:
  * Type in a weak password (i.e. doesn't match the password rules)
  * Type in a password you used previously
  * Type in a strong password (i.e. does match the password rules)
* Use the _Generate strong password_ button and make sure it works as expected

fixes #48404
